akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowWatchTerminationSpec.scala

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import akka.actor.Status.Failure
+import akka.pattern.pipe
+import akka.stream._
+import akka.stream.testkit.AkkaSpec
+import akka.stream.testkit.Utils._
+import akka.stream.testkit.scaladsl.{ TestSink, TestSource }
+
+import scala.util.control.NoStackTrace
+import scala.concurrent.duration._
+
+class FlowWatchTerminationSpec extends AkkaSpec {
+
+  val settings = ActorMaterializerSettings(system)
+
+  implicit val materializer = ActorMaterializer(settings)
+  implicit val ec = system.dispatcher
+
+  "A WatchTermination" must {
+
+    "complete future when stream is completed" in assertAllStagesStopped {
+      val (future, p) = Source(1 to 4).watchTermination().toMat(TestSink.probe[Int])(Keep.both).run()
+      p.request(4).expectNext(1, 2, 3, 4)
+      future.pipeTo(testActor)
+      expectMsg(())
```

Comment:
these should be `future.futureValue should ===(Done)` (which requires extending ScalaFutures and ConversionCheckedTripleEquals and adding an `implicit val patience = PatienceConfig(3.seconds)`)